### PR TITLE
Roles added, DB authentication configured

### PR DESCRIPTION
**Pull request by hpavlo**

### CODE REVIEW

File: src/main/java/mate/academy/spring/config/SecurityConfig.java [hunk 1]
```diff
@@ -1,27 +1,62 @@
...
-                .withUser("1@123").password(getEncoder().encode("123")).roles("USER");
+        auth.userDetailsService(userDetailsService).passwordEncoder(passwordEncoder);
     }
 
     protected void configure(HttpSecurity http) throws Exception {
         http
                 .authorizeRequests()
+                .antMatchers("/register").permitAll()
+                .antMatchers(HttpMethod.GET, "/cinema-halls")
+                    .hasAnyAuthority(ADMIN.name(), USER.name())
```
Comment (CyberDogFK) [inline src/main/java/mate/academy/spring/config/SecurityConfig.java:35]: You use everywhere has authority(), and you can change it to his role if in class UserDetailsServliseImpl you change setting .authorites to .roles

File: src/main/java/mate/academy/spring/service/impl/UserDetailsServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        Optional<User> userOptional = userService.findByEmail(username);
+        if (userOptional.isPresent()) {
+            User user = userOptional.get();
+            UserBuilder userBuilder = withUsername(username);
+            userBuilder.password(user.getPassword());
+            userBuilder.authorities(user.getRoles()
```
Comment (CyberDogFK) [inline src/main/java/mate/academy/spring/service/impl/UserDetailsServiceImpl.java:29]: Here you can change it to .roles and start work with roles

File: src/main/java/mate/academy/spring/config/SecurityConfig.java [hunk 3]
```diff
@@ -1,27 +1,62 @@
...
+                .antMatchers(HttpMethod.GET, "/orders")
+                    .hasAuthority(USER.name())
+                .antMatchers(HttpMethod.POST, "/orders/complete")
+                    .hasAuthority(USER.name())
+                .antMatchers(HttpMethod.PUT, "/shopping-carts/movie-sessions")
+                    .hasAuthority(USER.name())
+                .antMatchers(HttpMethod.GET, "/shopping-carts/by-user")
+                    .hasAuthority(USER.name())
+                .antMatchers(HttpMethod.GET, "/users/by-email")
+                    .hasAuthority(ADMIN.name())
```
Comment (CyberDogFK) [inline src/main/java/mate/academy/spring/config/SecurityConfig.java:59]: To make your code shorter, you can use responsibility to give to antMacher  array of addresses, but I don't know is a best practice

**Review by CyberDogFK — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/spring/dto/response/UserResponseDto.java [hunk 4]
```diff
@@ -1,8 +1,11 @@
 package mate.academy.spring.dto.response;
 
+import java.util.List;
+
 public class UserResponseDto {
     private Long id;
     private String email;
+    private List<Long> roleIds;
```
[the remark this answers is not in the document]
Comment (pavlogook) [reply src/main/java/mate/academy/spring/dto/response/UserResponseDto.java:8]: Thanks, but I don't see the point. There will still be an array in json)

File: src/main/java/mate/academy/spring/config/SecurityConfig.java (hunk 3 above)
Comment (pavlogook) [reply src/main/java/mate/academy/spring/config/SecurityConfig.java]: It's big, but I don't see any problems with reading it)))

File: src/main/java/mate/academy/spring/model/Role.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+import java.util.Objects;
+import javax.persistence.Entity;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+
+@Entity
+public class Role {
```
Comment (hronograf) [inline src/main/java/mate/academy/spring/model/Role.java:12]: let's add `Table` annotation and table name

File: src/main/java/mate/academy/spring/dto/response/UserResponseDto.java (hunk 4 above)
Comment (hronograf) [inline src/main/java/mate/academy/spring/dto/response/UserResponseDto.java:8]: if we use `Set` in model, let's use set in dto also

**Review by hronograf — APPROVED** [review]
Nice! 
@pavlogook check comments

**Pull request closed without merging**